packages/braid-design-system/src/lib/components/TooltipRenderer/TooltipRenderer.tsx

```diff
@@ -175,6 +206,24 @@ export const TooltipRenderer = ({
     },
   );
 
+  // If the tooltip is visible and the size or position of either the trigger
+  // or the tooltip has changed, then update the tooltip size and position.
+  if (
```

Comment:
Should this be wrapped in a `useEffect`, [or even `useLayoutEffect`](https://react.dev/reference/react/useLayoutEffect#measuring-layout-before-the-browser-repaints-the-screen)?

Reply:
Deliberately not, as not wanting to delay rendering the actual component, or have an effect that is driven by specific data changing.

The thinking being if this component is re-rendering, and the tooltip is already visible, check if it needs to be repositioned. Happy to discuss when you have time.

Reply:
In that case, this `if` block should be wrapped in `useLayoutEffect`, like so:

```ts
// If the tooltip is visible and the size or position of either the trigger
// or the tooltip has changed, then update the tooltip size and position.
useLayoutEffect(() => {
  if (
    controlledVisible &&
    update &&
    (doesBoundingBoxNeedUpdating(triggerRef, triggerBoundingBoxRef.current) ||
      doesBoundingBoxNeedUpdating(tooltipRef, tooltipBoundingRectRef.current))
  ) {
    triggerBoundingBoxRef.current = normaliseRect(
      triggerRef?.getBoundingClientRect(),
    );
    tooltipBoundingRectRef.current = normaliseRect(
      tooltipRef?.getBoundingClientRect(),
    );

    update();
  }
});
```

Notice how the dependency array is missing, which makes the code run on every render, which is equivalent to what you had before.
Since the code has a side-effect, it should be wrapped in `use*Effect`.

Reply:
Haha so it was less of a question then 😄 

Let's chat more about the right direction for the requirements and test it out.